packages/pluggableWidgets/line-chart-native/src/components/LineChart.tsx

```diff
@@ -0,0 +1,163 @@
+                                width={chartDimensions?.width}
+                                padding={style.chart?.padding}
+                                scale={
+                                    dataTypesResult
+                                        ? {
+                                              x: dataTypesResult.x === "number" ? "linear" : "time",
+                                              y: dataTypesResult.y === "number" ? "linear" : "time"
+                                          }
+                                        : undefined
+                                }
+                                style={style.chart}
+                            >
+                                <VictoryAxis
+                                    style={style.xAxis}
+                                    axisLabelComponent={
+                                        xAxisLabel ? (
+                                            <VictoryLabel
+                                                dx={style.xAxis?.axisLabel?.horizontalOffset}
+                                                dy={style.xAxis?.axisLabel?.verticalOffset}
+                                            />
+                                        ) : (
+                                            undefined
+                                        )
+                                    }
+                                    label={xAxisLabel}
+                                />
+                                <VictoryAxis
+                                    dependentAxis
+                                    style={style.yAxis}
+                                    axisLabelComponent={
+                                        yAxisLabel ? (
+                                            <VictoryLabel
+                                                dx={style.yAxis?.axisLabel?.verticalOffset}
+                                                dy={style.yAxis?.axisLabel?.horizontalOffset}
+                                            />
+                                        ) : (
+                                            undefined
+                                        )
+                                    }
+                                    label={yAxisLabel}
+                                />
+                                {chartLines}
+                            </VictoryChart>
+                        ) : null}
+                    </View>
+
+                    {showLegend ? <Legend style={style} series={series} /> : null}
+                </Fragment>
+            )}
+        </View>
+    );
+}
+
+function getDataTypes(series: LineChartSeries[]): { x: string; y: string } | Error | undefined {
+    let dataTypes: { x: string; y: string } | undefined;
+
+    for (const element of series) {
+        const { dataPoints } = element;
+
+        if (dataPoints.length > 0) {
+            const xDataType = typeof dataPoints[0].x;
```

Comment:
Can we do something smart here like `const [dataPoint] = dataPoints;`

Reply:
I am not seeing how. We need to know the typeof.

Reply:
I mean.. extract the variable and reuse in line 157 and 158

Reply:
Changed it.